google/cloud/storage/internal/rest_client.cc

```diff
@@ -104,11 +104,45 @@ Status AddAuthorizationHeader(Options const& options,
   return {};
 }
 
-RestClient::RestClient(google::cloud::Options options)
-    : storage_rest_client_(
-          rest::MakePooledRestClient(RestEndpoint(options), options)),
-      iam_rest_client_(
-          rest::MakePooledRestClient(IamRestEndpoint(options), options)),
+std::shared_ptr<RestClient> RestClient::Create(Options options) {
+  std::unique_ptr<rest::RestClient> storage_rest_client;
+  std::unique_ptr<rest::RestClient> iam_rest_client;
+  auto curl_options = options;
+  auto storage_endpoint = RestEndpoint(options);
+  auto iam_endpoint = IamRestEndpoint(options);
+
+  if (!options.has<AuthorityOption>() &&
+    absl::StrContains(storage_endpoint, "googleapis.com")) {
+    storage_rest_client = rest::MakePooledRestClient(
+        storage_endpoint,
+        options.set<AuthorityOption>("storage.googleapis.com"));
```

Comment:
Consider the case where `options == Options{}.set<IamEndpointOption>("foo").set<RestEndpointOption>("bar.googleapis.com")`

Then line 118 will set the `AuthorityOption` to `storage.googleapis.com` and that will get used in line 131.  Is that what we want?